diff --git a/webrtc-sys/build.rs b/webrtc-sys/build.rs
--- a/webrtc-sys/build.rs
+++ b/webrtc-sys/build.rs
@@ -127,1 +127,1 @@
-    println!("cargo:rustc-link-lib=static=webrtc");
+    println!("cargo:rustc-link-lib=dylib=webrtc");
@@ -171,1 +171,7 @@
-            add_gio_headers(&mut builder);
+            for lib_name in ["glib-2.0", "gio-2.0"] {
+                if let Ok(lib) = pkg_config::Config::new().cargo_metadata(false).probe(lib_name) {
+                    for path in lib.include_paths {
+                        builder.include(&path);
+                    }
+                }
+            }
